Bonsai.Configuration/ScriptExtensionsProvider.cs

```diff
@@ -143,15 +143,26 @@ string GetAssemblyLocation(string fileName)
                 return fileName;
             }
 
-            var compilerParameters = new CompilerParameters(assemblyReferences, assemblyFile);
-            compilerParameters.GenerateExecutable = false;
-            compilerParameters.GenerateInMemory = false;
-            compilerParameters.IncludeDebugInformation = includeDebugInformation;
+            var compilerParameters = new CompilerParameters(assemblyReferences, assemblyFile)
+            {
+                GenerateExecutable = false,
+                GenerateInMemory = false,
+                IncludeDebugInformation = includeDebugInformation,
+                CompilerOptions = "",
+            };
+
+            if (scriptEnvironment.GetAllowUnsafeBlocks())
+            {
+                compilerParameters.CompilerOptions += " /unsafe";
+            }
+            
             if (!includeDebugInformation)
             {
-                compilerParameters.CompilerOptions = "/optimize";
+                compilerParameters.CompilerOptions += " /optimize";
             }
 
+            compilerParameters.CompilerOptions = compilerParameters.CompilerOptions.TrimStart();
```

Comment:
I guess an alternative to this would be to have a list of strings and assign only once at the end by concatenating all options using white space as a separator. Might be useful if later we keep adding more compiler options.

Reply:
I almost wrote it that way but thought it'd come across as extra noise, plus the string command line options will probably go away with https://github.com/bonsai-rx/bonsai/issues/778 anyway. I'll change it to avoid the itch either way though.

Reply:
Done